server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -146,8 +146,11 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         if (event.nodesDelta().masterNodeChanged()) {
-            if (event.state().nodes().isLocalNodeElectedMaster()) {
-                metadataTracker.maybeStart();
+            if (event.state().nodes().isLocalNodeElectedMaster()
+                && event.previousState().nodes().isLocalNodeElectedMaster() == false) {
+                for (var subscriptionName : currentSubscriptionsMetadata.subscription().keySet()) {
+                    metadataTracker.startTracking(subscriptionName);
```

Comment:
Minor: The method `startTracking` is internally synchronized and creates a copy-on-write on each subscription. To make this more lightweight we could maybe have ` public boolean startTracking(Collection<String> subscriptionName)` to add all subscriptions at once.